test/e2e/e2e_tests_suite_test.go

```diff
@@ -0,0 +1,222 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+
+	"github.com/openshift-kni/commatrix/pkg/client"
+	"k8s.io/client-go/tools/clientcmd"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-kni/commatrix/pkg/consts"
+	"github.com/openshift-kni/commatrix/pkg/types"
+	"github.com/openshift-kni/commatrix/pkg/utils"
+
+	commatrixcreator "github.com/openshift-kni/commatrix/pkg/commatrix-creator"
+	"github.com/openshift-kni/commatrix/pkg/endpointslices"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("commatrix", func() {
+	It("should apply firewall by blocking all ports except the ones OCP is listening on", func() {
+		By("generating the commatrix")
+
+		kubeconfig, ok := os.LookupEnv("KUBECONFIG")
+		if !ok {
+			Fail("must set the KUBECONFIG environment variable")
+		}
+
+		config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+		Expect(err).NotTo(HaveOccurred())
+
+		oc := configv1client.NewForConfigOrDie(config)
+
+		cs, err := client.New()
+		Expect(err).NotTo(HaveOccurred())
+
+		deployment := types.MNO
+		isSNO, err := isSNOCluster(oc)
+		Expect(err).NotTo(HaveOccurred())
+
+		if isSNO {
+			deployment = types.SNO
+		}
+
+		infra := types.Cloud
+		isBM, err := infrType(oc)
+		Expect(err).NotTo(HaveOccurred())
+
+		if isBM {
+			infra = types.Baremetal
+		}
+
+		epExporter, err := endpointslices.New(cs)
+		Expect(err).ToNot(HaveOccurred())
+
+		commMatrix, err := commatrixcreator.New(epExporter, "", "", infra, deployment)
+		Expect(err).NotTo(HaveOccurred())
+
+		matrix, err := commMatrix.CreateEndpointMatrix()
+		Expect(err).NotTo(HaveOccurred())
+
+		masterMat, workerMat := matrix.SeparateMatrixByRole()
+
+		masterNFT, err := masterMat.ToNFTables()
+		Expect(err).NotTo(HaveOccurred())
+
+		By("apply firewall on master nodes")
+
+		err = applyRulesToNode(cs, "master", masterNFT)
+		Expect(err).ToNot(HaveOccurred())
+
+		if !isSNO {
+			By("apply firewall on worker nodes")
+
+			workerNFT, err := workerMat.ToNFTables()
+			Expect(err).NotTo(HaveOccurred())
+
+			err = applyRulesToNode(cs, "worker", workerNFT)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+})
+
+func applyRulesToNode(cs *client.ClientSet, role string, NFTtable []byte) error {
+	utilsHelpers := utils.New(cs)
+
+	nodesList, err := cs.CoreV1Interface.Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
```

Comment:
this function is in the tests folder you can just do Expect(err).ToNot(HaveOccurred())